src/main/scala/cromwell/binding/WdlNamespace.scala

```diff
@@ -266,25 +302,37 @@ case class WdlNamespace(ast: Ast, source: WdlSource, importResolver: ImportResol
    */
   def coerceRawInputs(rawInputs: WorkflowRawInputs): Try[WorkflowCoercedInputs] = {
 
-    def coerceRawInput(fqn: FullyQualifiedName, wdlType: WdlType): Try[WdlValue] = fqn match {
-      case _ if rawInputs.contains(fqn) =>
-        val rawInput = rawInputs.get(fqn).get
-        wdlType.coerceRawValue(rawInput).recoverWith {
-          case e => Failure(new UnsatisfiedInputsException(s"Failed to coerce input $fqn value $rawInput of ${rawInput.getClass} to $wdlType."))
+    def coerceRawInput(input: WorkflowInput): Try[Option[WdlValue]] = input.fqn match {
+      case _ if rawInputs.contains(input.fqn) =>
+        /* Workflow inputs can sometimes have multiple types.  For example, `${sep="," var+}`
+         * can accept a `String` or `Array[String]`.  This coercion tries to coerce to ALL valid types
+         * and just chooses the first one that works, or returns a Failure.
+         */
```

Comment:
@mcovarr I'm not opposed to that idea.  The only argument I have for keeping it is that it meets user expectations a little closer.  For example:

```
${sep=" " File x+}
```

If we're telling users that `x` only accepts an `Array[File]` then I think that's misleading because the type really is `File`.  The language I've always used when describing this in english is: "Treated as a regular parameter if you pass it a matching primitive type.  If given an array of that primitive type, it will expand it via the separator"

I'm trying to brainstorm some alternatives to the WDL syntax that appease both concerns:

```
${sep=" " Array[File] x}
```

Then, the rule can be "if `sep` is specified, append the elements in the array together into a string.  If not, then replace this variable with a `File` reference to the array values."

Seems okay.  Then we'd be getting rid of `?` and `*` altogether, which does simplify things a bit.  The current approach offers slightly better static checking because it *requires* `sep` to be specified if `?` or `*` are specified.

What do you think?